### DISCUSSION TITLE
nurturing a store manager

### DISCUSSION THREAD

**Discussion by valjusha** [solidjs/solid#735] [Q&A]
Hi, please tell me. Is it possible to build a data store over solidJS?
With a lot of data over a websocket and creating dynamic stores with subscriptions (any IoC/DI for business logic)? It is possible to use MobX, Effector or any?

I am only reading the documentation about data storage. It seems to me that all the functionality is strongly tied to the UI. I don’t need it

**Comment by atk** [comment]
Yes, you can use them, but you shouldn't (need to). The performance in Solid.js comes from the integration of a very efficient fine-grained reactive state management combined with a similarly efficient handling of DOM changes as side effects of that same state. Layering a different state manager (especially one that works with immutable objects) over it will only reduce performance.

At the same time, solid comes with powerful tools to manage state. Small state can be done with signals, complex state with stores - and nothing stops you from combining both.
